ccswbs/gus
# Let content authors add files via the WYSIWYG editor

# Summary of changes
Minimum Viable Product version of https://dev.azure.com/uoguelph/CCS/_workitems/edit/74912/ - content authors can now insert files via the existing Linkit tool and no longer have to manually copy and paste absolute links. These links are then processed by an enhanced text-parsing function to start with the correct base URL.

## Frontend
- Add media file to schema in `gatsby-node.js`
- Create `AnchorTag` function to process links, detecting if link is of type "media" and prepending the appropriate Drupal base URL
- Refactor `generalText.js` to move all text-processing functions to `ug-utils.js`
- Refactor `ug-utils.js` to use alphabetical ordering for functions and remove obsolete/unused code
- Use the new text-parsing functions in `accordion.js`, `mediaText.js` and `pageTabs.js`

### Spring Cleaning
- Remove unused `footer.js` component (not needed since we use the web components version)
- Remove unused `newsGrid.js` and `grid.js` components (leftovers from long-abandoned original program page template)
- Remove unused `navTabs.js` component and related files (this was replaced by `pageTabs.js` some time ago)
- Remove unused `relatedPages.js` component ([saved here](https://github.com/ccswbs/gus/wiki/Code-backup#code-from-relatedpagesjs) in case we want to resurrect it)
- Remove unused `sidebar.js` component (too old and out-of-date for repurposing)
- Remove calls to obsolete and unnecessary `contentExists` function from the following files:
    - `footerOVC.js`
    - `heroVideo.js`
    - `stats.js`
- Delete `contentExists` function from `ug-utils.js`

[ x ] My changes are accessible (at minimum WCAG 2.0 Level AA)
[ x ] My changes are responsive and appear as expected on mobile and desktop views.
[ x ] After merging, I will update the [Content Hub documentation](https://uoguelphca.sharepoint.com/sites/UniversityContentHubInformationGroup) and ensure the Content Hub trainer understands how my changes will affect users.

## Backend
See [this message](https://teams.microsoft.com/l/message/19:f2ae5b00dc95407d8902b0e04ff98ec1@thread.tacv2/1711484549308?tenantId=be62a12b-2cad-49a1-a5fa-85f4f3156a7d&groupId=cde949b1-e9d8-483b-8d20-e9b8fd9e493d&parentMessageId=1711484549308&teamName=Content%20Hub%20Team&channelName=Multidev%20Merge%20Requests&createdTime=1711484549308&allowXTenantAccess=false)  for details

# Test Plan
- Go to https://tqfilefix--test-ugconthub.netlify.app/file-test/ and verify all links work as expected
- Go to https://tqfilefix--test-ugconthub.netlify.app/international/chat/ and verify the external script (chat functionality) loads successfully
- Go to https://tqfilefix--test-ugconthub.netlify.app/ovc/experts/ (or any other OVC page) and verify the custom OVC footer displays as expected
- Go to https://tqfilefix--test-ugconthub.netlify.app/oac/study/ and verify stats display as expected
- Go to https://tqfilefix--test-ugconthub.netlify.app/studentexperience/ and verify the hero video plays successfully
- Go to https://tq-filefix-chug.pantheonsite.io/ and create a basic page (or clone https://tq-filefix-chug.pantheonsite.io/file-test)
- Add a link to a file using the Linkit tool
- Verify the link is inserted in the Drupal Media File format (i.e. with relevant data-entity attributes)
- Run a local build with your `DRUPAL_BASERUL` env variable set to https://tq-filefix-chug.pantheonsite.io/
- Verify the URL to the file you linked starts with https://tq-filefix-chug.pantheonsite.io/ 

## Known issues
- The `AnchorTag` function as written cannot handle edge cases where an anchor tag has inline styles so instead it strips them out and leaves the rest of the link intact. Since inline styles are strongly discouraged anyway and break accessibility rules, we could leave this as-is.
